pkg/domain/infra/runtime_libpod.go

```diff
@@ -126,12 +127,17 @@ func getRuntime(ctx context.Context, fs *flag.FlagSet, opts *engineOpts) (*libpo
 		storageSet = true
 	}
 
-	if fs.Changed("root") {
+	gr, _ := fs.GetString("graph-root")
+	if fs.Changed("root") || len(gr) > 0 {
+		options = append(options, libpod.WithNewDB())
 		storageSet = true
 		storageOpts.GraphRoot = cfg.Engine.StaticDir
 		storageOpts.GraphDriverOptions = []string{}
 	}
-	if fs.Changed("runroot") {
+
+	rr, _ := fs.GetString("run-root")
+	if fs.Changed("runroot") || len(rr) > 0 {
```

Comment:
Where else is rr being used?

Reply:
rr is just being used below in `if fs.Changed("runroot") || len(rr) > 0 {` the issue is I cannot just do `fs.Changed` since the flag will not always exist at runtime.